samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs

```diff
@@ -64,5 +67,95 @@ public async Task<IEnumerable<EducationAssignment>> GetMeAssignmentsFromNonArchi
+        /// <summary>
+        /// Workflow to get assignments evolvable enums
+        /// </summary>
+        /// <param name></param> 
+        public async Task AssignmentsEvolvableEnums()
+        {
+            try
+            {
+                int retries = 0;
+                string assignmentId = string.Empty;
+                string submissionId = string.Empty;
+
+                // Get a Graph client using delegated permissions
+                var graphClient = MicrosoftGraphSDK.GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], _config["teacherAccount"], _config["password"]);
+
+                // Create a draft assignment with displayName = "Verify assignments states [inactive]"
+                var assignmentInactive = await MicrosoftGraphSDK.Assignment.CreateAsync(graphClient, _config["classId"]);
+                assignmentId = assignmentInactive.Id;
+                Console.WriteLine($"Assignment created successfully {assignmentInactive.Id} in state {assignmentInactive.Status}");
+
+                // Create a draft assignment with displayName = "Verify assignments states [assigned]"
+                var assignmentAssigned = await MicrosoftGraphSDK.Assignment.CreateAsync(graphClient, _config["classId"]);
+                Console.WriteLine($"Assignment created successfully {assignmentAssigned.Id} in state {assignmentAssigned.Status}");
+
+                // Create a draft assignment with displayName = "Verify assignments states [draft]"
+                var assignmentDraft = await MicrosoftGraphSDK.Assignment.CreateAsync(graphClient, _config["classId"]);
+                Console.WriteLine($"Assignment created successfully {assignmentDraft.Id} in state {assignmentDraft.Status}");
+
+                // Teacher publishes the assignment to make it appears in the student's list
+                assignmentInactive = await MicrosoftGraphSDK.Assignment.PublishAsync(graphClient, _config["classId"], assignmentId);
+                Console.WriteLine($"Assignment {assignmentInactive.Id} publish in process");
+
+                // Verify assignment state, publish is completed until state equals "Assigned"
+                while (assignmentInactive.Status != EducationAssignmentStatus.Assigned && retries <= MAX_RETRIES)
+                {
+                    // Print . in the log to show that the call is being retried
+                    Console.WriteLine(".");
+
+                    assignmentInactive = await MicrosoftGraphSDK.Assignment.GetAssignmentAsync(graphClient, _config["classId"], assignmentId);
+
+                    // If you are calling this code pattern in Backend agent of your service, then you want to retry the work after some time. The sleep here is just an example to emulate the delay
+                    Thread.Sleep(2000);
+                    retries++;
+                }
+                Console.WriteLine($"Assignment {assignmentInactive.Id} publish is completed. Status: {assignmentInactive.Status}");
```

Comment:
please move all the PUBLISH code into a new method at the **GlobalMethods** class.
It is a repetitive code across different samples, we can have it in a global mehtod and just call it and pass the assignment object as parameter.

Reply:
Updated the method in Global